File: lc_py/codeforces/AB.py
def solve(s: str, c: str) -> int:
    pos = [i for i, ch in enumerate(s) if ch == c]
    if not pos:
        return 0
    shifted = [p - i for i, p in enumerate(pos)]
    shifted.sort()
    m = shifted[len(shifted) // 2]
    return sum(abs(x - m) for x in shifted)

def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        s = input().strip()
        ans = min(solve(s, 'a'), solve(s, 'b'))
        print(ans)

if __name__ == "__main__":
    main()


'''
让所有 'a'（或 'b'）聚成一块连续的区间
假设a的出现位置pos = [p1, p2, p3, ..., pk]
把它们排成一个连续区间 [x, x+1, ..., x+k-1] 第 i 个 'a' 希望被放到新位置 x + (i - 1)
                cost(x)=i=1∑k ​∣pi​−(x+i−1)∣ -> |pi - (i - 1) - x| -> |qi - x| x的qi中位数

'''

File: lc_go/codeforces/string/ab.go
package main

import (
	"fmt"
	"slices"
)

func abs(x int64) int64 {
	if x < 0 {
		return -x
	}
	return x
}

func solve(s []byte, c byte) int64 {
	pos := []int{}
	for i, b := range s {
		if b == c {
			pos = append(pos, i)
		}
	}
	if len(pos) == 0 {
		return 0
	}

	shifted := make([]int64, len(pos))
	for i := range pos {
		shifted[i] = int64(pos[i] - i)
	}
	slices.Sort(shifted)
	m := shifted[len(shifted)/2]

	var res int64
	for _, x := range shifted {
		res += abs(x - m)
	}
	return res
}

func min(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

func main() {
	var t int
	fmt.Scan(&t)
	for i := 0; i < t; i++ {
		var n int
		var s string
		fmt.Scan(&n, &s)
		fmt.Println(min(solve([]byte(s), 'a'), solve([]byte(s), 'b')))
	}
}
